### ISSUE TITLE
Datasource option MYSQL is missing i 3.x for MacOs

### ISSUE THREAD

**Issue by janpalmen** [qgis/QGIS#34540]
**Describe the bug**
Datasource option MYSQL is missing in QGIS 3.x for MacOs. 

**How to Reproduce**
Add vector layer, click database option. Select database type and read the list. MYSQL is missing.

**QGIS and OS versions**
3.4 and 3.10 on macOs 10.15 Catalina

**Additional context**
MySQL 8.0.19

**Comment by maxime400** [comment]
Hi, i have the same issue, did you manage to add Mysql in database source type ?

**Comment by OfflineVibe** [comment]
Hi, have you managed to find a solution? I am having the same problem.

**Comment by janpalmen** [comment]
No, I have not found a solution. Latest QGIS version I have tried, where this problem persists, is 3.12.

**Comment by chrisk8er** [comment]
still waiting for the solution...

**Comment by gioman** [comment]
> still waiting for solution...

@chrisk8er don't expect it to materialize out from thin air, especially given the general lack of interest around support for MySQL in QGIS.

**Comment by PeterPetrik** [comment]
I consider this as an enhancement, not a bug. Since it is not a bug with the existing packaging, just missing functionality which lack of general interest/funding as @gioman mentioned

**Comment by JannePalmen** [comment]
Yes, I agree. Since the MySQL specific instructions now is removed from the documentation about data sources (https://docs.qgis.org/3.16/en/docs/user_manual/managing_data_source/index.html) it is no longer a Bug but a Featured Request.

I made a workaround on my Mac using a geojson connection in GQIS. I used Apache webserver to parse a php page that fetched data from MySQL and created a geojson-output that QGIS could read. Works like a charm.

**Comment by olsen232** [comment]
I can see that QGIS devs have to spend effort where it is most rewarded, and so I can broadly agree with the decision to consider this a feature-request with a "lack of interest/funding". But, I think whoever removed it from the 3.x MacOS packaging would have done better to put a little note about it in the documentation, instead of just deleting all references to it. This bug / feature request is the only semi-official documentation I have found so far (perhaps I am googling wrong)

**Comment by nyalldawson** [comment]
@olsen232 to clarify, it was never removed from the all-in-one mac packages. Rather it has **never** been supported in those installers. The alternative installers from https://www.kyngchaos.com/software/qgis/ may have mysql enabled, but I'm not sure about that (@kyngchaos  can you confirm?)

**Comment by olsen232** [comment]
I can now confirm - I just installed the following:
http://www.kyngchaos.com/files/software/qgis/QGIS-1.8.0-2.dmg
which had the following dependencies:
http://www.kyngchaos.com/files/software/frameworks/GDAL_Complete-1.9.dmg
http://www.kyngchaos.com/files/software/frameworks/GSL_Framework-1.16-1.dmg
(and perhaps others that I may have already had installed by chance)

And it does indeed have the option of adding a MySQL layer. Posting here in case it helps someone, although it's a bit of a niche interest at this point.

Apparently the QGIS-1.8.0-2's MySQL client authentication isn't compatible with MySQL server. But that's another story.

**Comment by kyngchaos** [comment]
Long ago I had MySQL built into GDAL.  I removed it because I wanted to keep the GDAL framework free of GPL the license.  I was going to add a MySQL GDAL plugin, but never got around to it, and nobody complained.

I've been thinking about it lately, but haven't had time to work on it.  This would only affect my framework-based QGIS package, I don't have anything to do with the all in one package.

**Comment by PeterPetrik** [comment]
it looks like MySQL is compiled and packaged in the all-in-one installer, only for some reason GDAL or QGIS provider is not correctly compiled (https://github.com/qgis/QGIS-Mac-Packager/blob/master/qgis_deps/recipes/mysql/recipe.sh)
